x-pack/test/security_functional/plugins/test_endpoints/server/init_routes.ts

```diff
@@ -39,6 +39,32 @@ export function initRoutes(
   );
 
   const router = core.http.createRouter();
+
+  for (const isAuthFlow of [true, false]) {
+    router.get(
+      {
+        path: `/authentication/app/${isAuthFlow ? 'auth_flow' : 'not_auth_flow'}`,
```

Comment:
> Steps to test
> Use existing testing configs to run Kibana with different configurations:

> 1. Login selector is enabled
> node scripts/functional_tests_server.js --config x-pack/test/security_functional/login_selector.config.ts

> 2. Login selector is disabled, login page is available, but not default == we ignore the existence of /login UI
> node scripts/functional_tests_server.js --config x-pack/test/security_api_integration/saml.config.ts

> 3. Login selector is not enabled, but login page is available and default
> node scripts/functional_tests_server.js --config x-pack/test/security_api_integration/http_bearer.config.ts

__question:__ would it make sense to add an automated test into each of these suites that replicates what we do to manually test these changes?

```
// PSEUDOCODE
it('handing unexpected post-auth errors', () => ...

/authentication/app/not_auth_flow --> expect(200, 'rendered text') in all test suites
/authentication/app/not_auth_flow?statusCode=400 --> expect(400, JSON error) in all test suites
/authentication/app/not_auth_flow?statusCode=401 --> expect(user-friendly UI) OR expect(302, /logout) depending on the test suite

/authentication/app/auth_flow --> expect(200, 'rendered text') in all test suites
/authentication/app/auth_flow?statusCode=400 --> expect(user-friendly UI) OR expect(302, /logout) depending on the test suite
/authentication/app/auth_flow?statusCode=401 --> expect(user-friendly UI) OR expect(302, /logout) depending on the test suite
```